src/borg/repository.py

```diff
@@ -1475,6 +1476,20 @@ def delete_segment(self, segment):
         except FileNotFoundError:
             pass
 
+    def clear_empty_dirs(self):
+        "Delete empty data dirs, i.e those with no segment files"
+        data_dir = os.path.join(self.path, 'data')
+        segment_dirs = [f.path for f in os.scandir(data_dir) if f.is_dir()]
```

Comment:
see the `os.listdir` calls that work on the data_dir, they also check the subdir name for isdigit - you should also do that here.

also, instead of creating the full list here, you could just set up a generator.